packages/deployments/contracts/contracts/core/connext/facets/ProposedOwnableFacet.sol

```diff
@@ -245,12 +264,61 @@ contract ProposedOwnableFacet is BaseConnextFacet, IProposedOwnable {
     _setOwner(s._proposed);
   }
 
-  function pause() public onlyOwner {
+  /**
+   * @notice Use to assign an address Router role
+   * Address with Router has permission to add new router
+   * and assign addresses as Role.Router
+   * Can only be called by Owner or Role.Router
+   * @dev requested address will be whitelisted as Role.Router under roles mapping
+   * @param _router - The address to be assigned as Role.Router under roles
+   */
+  function assignRoleRouter(address _router) public onlyOwnerOrRouter {
```

Comment:
Under this construction, the same EOA can't have multiple roles (i.e. be a router AND a watcher)... is that okay? Is that what we want? Is there a good reason as to why a singular EOA shouldn't be able to have multiple roles? (Seems like it's just inconvenient, but idk)

Reply:
Also, it seems weird that a router can just add any other arbitrary router (since the permissions are `onlyOwnerOrRouter`)? Is that normal?

Reply:
I think it makes sense to keep, a 1-1 relation between address and role.

Hmm, I think it would make more sense if the permission to add router falls under admin only.